sql/table.go

```diff
@@ -214,10 +214,9 @@ func encodeTableKey(b []byte, val parser.Datum) ([]byte, error) {
 	return nil, fmt.Errorf("unable to encode table key: %T", val)
 }
 
-func makeIndexKeyVals(desc *TableDescriptor,
-	index IndexDescriptor) ([]parser.Datum, error) {
-	vals := make([]parser.Datum, len(index.ColumnIDs))
-	for i, id := range index.ColumnIDs {
+func makeIndexKeyVals(desc *TableDescriptor, columnIDs []ColumnID) ([]parser.Datum, error) {
```

Comment:
> The caller doesn't have have a slice of columns. It has a descriptor and slice of column ids. I suppose we could push the work of creating a slice of columns to the caller, but it seems easy enough to do here.

Good enough for me.

> This function is only used in the context of indexes. Why should Index be dropped from the name?

Just because it doesn't expressly know about indexes; it's really transforming a slice of column ids into a slice of datums.